libraries/dotnet/EnergyOrigin.TokenValidation/EnergyOrigin.TokenValidation/B2C/TermsRequirementHandler.cs

```diff
@@ -0,0 +1,31 @@
+using EnergyOrigin.TokenValidation.b2c;
+using Microsoft.AspNetCore.Authorization;
+
+namespace EnergyOrigin.TokenValidation.B2C;
+
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+public class DisableTermsRequirementAttribute : Attribute;
+
+public class TermsRequirement : IAuthorizationRequirement;
+
+public class TermsRequirementHandler : AuthorizationHandler<TermsRequirement>
+{
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, TermsRequirement requirement)
+    {
+        var endpoint = context.Resource as Microsoft.AspNetCore.Http.Endpoint;
+        var disableTermsAttribute = endpoint?.Metadata.GetMetadata<DisableTermsRequirementAttribute>();
+
```

Comment:
We need to check if it's 3rd party or not. If 3rd party they dont have terms accepted.

Reply:
That should not be relevant. The requirement for having accepted terms, is only applied for the policy for users. So unless someone uses a "user-policy" for authorizing clients, in their controller actions, then i don't see a problem